server/node_filter.go

```diff
@@ -436,7 +436,7 @@ func (nf *NodeFilter) KubeletOutdatedNodes() (nodes []*cke.Node) {
 			log.Warn("kubelet's container runtime cannot be changed", nil)
 		case cke.KubernetesImage.Name() != st.Image:
 			fallthrough
-		case !reflect.DeepEqual(currentConfig, runningConfig):
+		case !equality.Semantic.DeepEqual(currentConfig, runningConfig):
```

Comment:
I use `equality.Semantic.DeepEqual` instead of `reflect.DeepEqual`.

`reflect.DeepEqual` has got false positives after upgrading k8s 1.23.
And the mtest never succeed with following log.
I dould not detect the root cause of the false positives.

```
2022-08-02T09:25:53.7562730Z [31mAug 02 09:18:45 host1 docker[1326]: 2022-08-02T09:18:45.409510Z host1 cke debug: "kubelet outdated" config="&TypeMeta***Kind:,APIVersion:,***" current_builtin_args="[kubelet --config=/etc/kubernetes/kubelet/config.yml --kubeconfig=/etc/kubernetes/kubelet/kubeconfig --hostname-override=10.0.0.101 --network-plugin=cni --container-runtime=remote --container-runtime-endpoint=unix:///var/run/k8s-containerd.sock]" current_builtin_env="map[]" current_extra_args="[]" current_extra_env="map[]" diff="" node="10.0.0.101" st_builtin_args="[kubelet --config=/etc/kubernetes/kubelet/config.yml --kubeconfig=/etc/kubernetes/kubelet/kubeconfig --hostname-override=10.0.0.101 --network-plugin=cni --container-runtime=remote --container-runtime-endpoint=unix:///var/run/k8s-containerd.sock]" st_builtin_env="map[]" st_extra_args="[]" st_extra_env="map[]"
```

Reply:
This is just speculation.
"Nullify TypeMeta" code blocks in `op/status.go` might have failed to run due to the change of the API version.
https://github.com/cybozu-go/cke/blob/v1.22.9/op/status.go#L129
https://github.com/cybozu-go/cke/blob/v1.22.9/op/status.go#L155
https://github.com/cybozu-go/cke/blob/v1.22.9/op/status.go#L182

Whether the speculation above is correct or not, we might be able to delete "Nullify TypeMeta" steps by using `equality.Semantic.DeepEqual()`.

Reply:
The "Nullify TypeMeta" in `op/status.go` is necessary because the TypeMeta in the `currentConfig` is also deleted.

https://github.com/cybozu-go/cke/blob/v1.22.9/cluster.go#L159
https://github.com/cybozu-go/cke/blob/v1.22.9/cluster.go#L207
https://github.com/cybozu-go/cke/blob/v1.22.9/cluster.go#L267

By the way, I forgot to update comments in `op/status.go`. Please review it. 4427ae4ddbbe61cc3c2efe24bf1ca82ed6daf9ab